dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java

```diff
@@ -0,0 +1,101 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * This DataFetcher returns a {@link HTMLPageAsset} given an URL. It also takes optional parameters
+ * to find a specific version of the page: languageId and pageMode.
+ *
+ * The returned page includes extra properties set by a page transformer.
+ *
+ */
+public class TemplateDataFetcher implements DataFetcher<Map<Object, Object>> {
+    @Override
+    public Map<Object, Object> get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            final User user = ((DotGraphQLContext) environment.getContext()).getUser();
+            final HttpServletRequest request = ((DotGraphQLContext) environment.getContext())
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)